client/js/main.ts

```diff
@@ -201,59 +259,113 @@ function loadAttendees (filter: string = queryField.value, checkedIn: string = c
-		if (!attendeeList.firstChild || numberOfExistingNodes < response.length) {
+
+		if (!attendeeList.firstChild || numberOfExistingNodes < attendees.length) {
 			// First load, preallocate children
 			status.textContent = "Preallocating nodes...";
-			for (let i = numberOfExistingNodes; i < response.length; i++) {
+			for (let i = numberOfExistingNodes; i < attendees.length; i++) {
 				let node = document.importNode(attendeeTemplate.content, true) as DocumentFragment;
 				node.querySelector("li")!.style.display = "none";
 				node.querySelector(".actions > button")!.addEventListener("click", checkIn);
 				attendeeList.appendChild(node);
 			}
 			(<any> window).mdc.autoInit();
-			console.warn(`Allocated ${response.length - numberOfExistingNodes} nodes due to insufficient number`);
+			console.warn(`Allocated ${attendees.length - numberOfExistingNodes} nodes due to insufficient number`);
 			status.textContent = "Loading...";
 		}
 
 		// Reuse nodes already loaded from template
 		let existingNodes = document.querySelectorAll("#attendees li") as NodeListOf<HTMLElement>;
 		for (let i = 0; i < existingNodes.length; i++) {
-			let attendee = response[i];
+			let attendee = attendees[i];
 			if (!!attendee) {
 				existingNodes[i].style.display = "";
 
-				existingNodes[i].id = "item-" + attendee.id;
-				existingNodes[i].querySelector("#name")!.textContent = attendee.name;
-
-				let emails = attendee.emails.reduce((prev, current) => {
-					if (prev.indexOf(current) === -1) {
-						prev.push(current);
-					}
-					return prev;
-				}, <string[]> []);
-				existingNodes[i].querySelector("#emails")!.textContent = emails.join(", ");
+				existingNodes[i].id = "item-" + attendee.user.id;
+				existingNodes[i].querySelector("#name")!.textContent = attendee.user.name;
+				existingNodes[i].querySelector("#emails")!.textContent = attendee.user.email;
 
 				let button = existingNodes[i].querySelector(".actions > button")!;
 				let status = existingNodes[i].querySelector(".actions > span.status")!;
-				let date = attendee.tags[tag].checked_in_date;
-				if (date) {
+
+				// Determine if user has the current tag
+				let tagInfo: IGraphqlTag[] = attendee.tags.filter((curr) => {
+					return curr.tag.name == tag;
+				});
+
+				if (tagInfo.length > 0 && tagInfo[0].checked_in) {
 					button.textContent = "Uncheck in";
 					button.classList.add("checked-in");
-					status.innerHTML = statusFormatter(date, attendee.tags[tag].checked_in_by);
+
+					let date = tagInfo[0].checked_in_date;
+					if (date && tagInfo[0].checked_in_by) {
+						status.innerHTML = statusFormatter(date, tagInfo[0].checked_in_by);
+					}
 				}
 				else {
 					button.textContent = "Check in";
 					button.classList.remove("checked-in");
 					status.textContent = "";
 				}
+				if (attendee.user.questions) {
+					let registrationInformation = attendee.user.questions.map(info => {
+						return info.name + ": " + info.value;
```

Comment:
Please change to use template strings (i.e. `${info.name}: ${info.value}`). Can also be simplified to 1 line without braces